packet/HardwareReservation.py

```diff
@@ -44,6 +44,12 @@ def __init__(self, data, manager):
         # except (KeyError, IndexError):
         #     self.attached_to = None
 
+
+        # not attached
+        if data["device"] is None:
+            self.attached_to = None
```

Comment:
Alternatively, we could use `except (KeyError, IndexError, TypeError):` in the following `try` block.

Reply:
@boz I think I would prefer that approach.  The overfit error handling is mildly concerning, but I don't see any risks that are not already present in the previous `try` block.

Reply:
Sure, will do.

I it that way at first but I felt like it could shadow other problems too easily.